bootstrap.py

```diff
@@ -74,14 +72,35 @@
 
 # ---- Install sub repos
 
+from install_dev_repos import REPOS, install_repo
+from spyder import get_versions
+from spyder.config.base import get_conf_path
+from spyder.utils import vcs
+
 installed_dev_repo = False
 if not args.no_install:
+    prev_branch = None
+    boot_branch_file = Path(get_conf_path("boot_branch.txt"))
+    if boot_branch_file.exists():
+        prev_branch = boot_branch_file.read_text()
+
+    revision, branch = vcs.get_git_revision(Path(__file__).parent)
+    boot_branch_file.write_text(branch)
+
+    logger.info("Previous branch: %s; current branch: %s", prev_branch, branch)
+
+    if "master" in (branch, prev_branch) and branch != prev_branch:
```

Comment:
Maybe the Spyder editable reinstall should be forced even if the branch is not `master`. Thinking about it is possible that a branch is based on `master` or `5.x` without being exactly `master` or `5.x` right?

Reply:
I suppose what matters is whether the parent commit has changed to/from the master branch. Pushed up a change.